### PR TITLE
fix(client): prevent lower jaw breaking on code evaluation

### LINKED ISSUE #46062: Entering new code after checking a valid solution breaks the lower jaw
### Describe the Issue

The new challenges in https://www.freecodecamp.org/learn/2022/responsive-web-design/ display the wrong UI in certain situations

### Affected Page

All project steps e.g. https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-2  

### Steps to Reproduce

1. Go to https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-2
2. Complete the step by adding `<h2>Cat Photos</h2>`
3. Click Check your code
4. Go to the editor and type anything
5. See image
6. Type another character
7. See image

### Expected behavior

The editor should say "Submit and go to the next challenge" until the user resets the step.

### PR DESCRIPTION
**Pull request by SadmanSyfe**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #46062

I created a variable that gets the value from 'challengeIsCompleted' when 'challengeIsCompleted' variable updates from the 'check code' button press. After that, I replaced 'challengeIsCompleted' with that variable everywhere 'challengeIsCompleted' is used.

### CODE REVIEW

**Comment by raisedadead** [comment]
As a new contributor, we encourage you to read our contributing guidelines. 

We expect our contributors to be aware of the process specific to this project. Following them religiously earns you the respect of fellow maintainers and saves everyone time.

Some examples of this are:

1. Make sure you are not editing off of GitHub (while you can, it's not a good idea).
2. Makes sure you actually follow the PR checklist and not just tick things off; we won't take you seriously otherwise.
3. Use the correct way to link PRs in the description of the PR by updating the `XXXXXX`.
4. ... and similar things.

We appreciate you taking the time to help us, and we hope to see more contributions from you.

Cheers & Happy Coding!

**Comment by SadmanSyfe** [comment]
@raisedadead ,
I have updated the title and description. It should be easier to review the PR.

**Comment by raisedadead** [comment]
Hi @SadmanSyfe 

Thanks for updating the title.

Have you gotten a chance to review our contribution guidelines? Especially https://contribute.freecodecamp.org/#/how-to-open-a-pull-request

I am deliberately not giving away exactly what needs to be changed to ensure that you have time to read the guide. 

Please note it's not about getting things done correctly. It's about getting things done the right way. We are here to help you help us :)

Cheers and happy coding.

**Comment by raisedadead** [comment]
I have gone ahead and made edits. But here is what happens when contributors do not follow recommendations. We ideally would like to avoid noise when we can.


image

**Comment by raisedadead** [comment]
Nice work so far @SadmanSyfe Do you mind checking and fixing the lining errors?

**Comment by raisedadead** [comment]
@SadmanSyfe what's up with the force pushes? Are you still working on this?

**Comment by raisedadead** [comment]
Thanks for confirming. 

You do not need to do that (in fact it causes noise by pinging a lot of people) unless you are adding changes. 

You only need to update a PR when:

1. You want to add new changes you forgot or think were missed.
2. Someone asked you to rebase the PR for example due to merge conflicts.

For the most part, because we use squash or fast-forward merge PRs will land just as they are intended. In simpler terms, the final changes you see on the "files" tab are what will appear on the main (or the PR target) branch when a PR is merged.

Thanks.

File: client/src/templates/Challenges/classic/lower-jaw.tsx [hunk 1]
```diff
@@ -37,6 +37,16 @@ const LowerJaw = ({
   const { t } = useTranslation();
   const submitButtonRef = React.createRef<HTMLButtonElement>();
   const testFeedbackRef = React.createRef<HTMLDivElement>();
+  // Wrapper for changing "test-status" only when test code is pressed
+  const [challengeIsCompletedWrapper, setchallengeIsCompletedWrapper] =
+    useState(false);
+
+  const tryToExecuteChallengeWrapper = () => {
```
Comment (ahmadabdolsaheb) [inline client/src/templates/Challenges/classic/lower-jaw.tsx:44]: I am not sure if we would need this.

So we could basically look for changes to `challengeIsCompleted` prop in a useEffect. Once it is true, we can persist its value in a state and called it `challengeHasBeenCompleted` or something more descriptive. 

The render the related elements accordingly.

**Review by ahmaxed — DISMISSED** [review]

File: client/src/templates/Challenges/classic/lower-jaw.tsx [hunk 2]
```diff
@@ -67,13 +69,22 @@ const LowerJaw = ({
...
       }, 500);
     }
 
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [challengeHasBeenCompleted]);
+
+  useEffect(() => {
+    if (!challengeHasBeenCompleted) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      setChallengeHasBeenCompleted(challengeIsCompleted!);
```
Comment (nhcarrigan) [inline client/src/templates/Challenges/classic/lower-jaw.tsx:85]:
```suggestion
      setChallengeHasBeenCompleted(!!challengeIsCompleted);
```
Rather than disabling the linter, can we convert it?

**Review by naomi-lgbt — DISMISSED** [review]

File: client/src/templates/Challenges/classic/lower-jaw.tsx (hunk 2 above)
Comment (ahmadabdolsaheb) [reply client/src/templates/Challenges/classic/lower-jaw.tsx]: good catch. added a check if `challengeIsCompleted` is truthy before setting `challengeHasBeenCompleted`

**Review by ahmaxed — APPROVED** [review]

**Review by Sembauke — APPROVED** [review]

**Comment by Sembauke** [comment]
Hi @[SadmanSyfe](https://github.com/SadmanSyfe) Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝
